ui/src/shared/components/LayoutCell.js

```diff
@@ -30,22 +34,39 @@ class LayoutCell extends Component {
     this.props.onSummonOverlayTechnologies(cell)
   }
 
+  grabDataForDownload = celldata => {
+    this.setState({celldata})
+  }
+
+  handleCSVDownload = cell => () => {
+    const joinedName = cell.name.split(' ').join('_')
+    const {celldata} = this.state
+    try {
+      download(dashboardtoCSV(celldata), `${joinedName}.csv`, 'text/plain')
+    } catch (error) {
+      errorThrown(error, 'Unable to download .csv file')
+      console.error(error)
+    }
+  }
+
   render() {
     const {cell, children, isEditable} = this.props
 
-    const {isDeleting} = this.state
+    const {isDeleting, celldata} = this.state
     const queries = _.get(cell, ['queries'], [])
 
     return (
       <div className="dash-graph">
         <LayoutCellMenu
           cell={cell}
+          dataExists={celldata.length}
```

Comment:
Very minor thing here. `dataExists` would probably be better served as a boolean instead of a number with `!!celldata.length`